### ISSUE TITLE
java.lang.NoClassDefFoundError: fastparse/core/Parser when running in Ammonite

### ISSUE THREAD

**Issue by RomanIakovlev** [alonsodomin/cron4s#162]
This happens when I run anything from Cron4s inside Ammonite REPL (http://ammonite.io/#Ammonite-REPL). I presume, this is a binary incompatibility between fastparse (http://www.lihaoyi.com/fastparse/) modules used in Ammonite and in Cron4s. I've tried to find which exactly versions of fastparse are used on both sides, but, unfortunately, this dependency seems to be transitive for both Ammonite and Cron4s, and finding exact one is rather challenging (Ammonite uses Mill as build tool, and I'm completely unfamiliar with it).

Here's the full Ammonite session illustrating this:

```
Loading...
Welcome to the Ammonite Repl 1.6.6-6-7e89968
(Scala 2.12.8 Java 1.8.0_162)
If you like Ammonite, please support our development at www.patreon.com/lihaoyi
@ import $ivy.`com.github.alonsodomin.cron4s::cron4s-core:0.5.0`
import $ivy.$

@ import cron4s._
import cron4s._

@ val parsed = Cron.unsafeParse("0 0 6 * * ?")
java.lang.NoClassDefFoundError: fastparse/core/Parser
  cron4s.Cron$.parse0(Cron.scala:80)
  cron4s.Cron$.parse(Cron.scala:50)
  cron4s.Cron$.unsafeParse(Cron.scala:74)
  ammonite.$sess.cmd2$.<init>(cmd2.sc:1)
  ammonite.$sess.cmd2$.<clinit>(cmd2.sc)
java.lang.ClassNotFoundException: fastparse.core.Parser
  java.net.URLClassLoader.findClass(URLClassLoader.java:381)
  ammonite.runtime.SpecialClassLoader.findClass(ClassLoaders.scala:241)
  java.lang.ClassLoader.loadClass(ClassLoader.java:424)
  java.lang.ClassLoader.loadClass(ClassLoader.java:357)
  cron4s.Cron$.parse0(Cron.scala:80)
  cron4s.Cron$.parse(Cron.scala:50)
  cron4s.Cron$.unsafeParse(Cron.scala:74)
  ammonite.$sess.cmd2$.<init>(cmd2.sc:1)
  ammonite.$sess.cmd2$.<clinit>(cmd2.sc)
```

Cron4s version is 0.5.0. Ammonite version is 1.6.6-6-7e89968.

**Comment by alonsodomin** [comment]
hi @RomanIakovlev, cron4s 0.5 uses FastParse 1.0 - which has been superseded by FastParse 2.0, that's where I think you will find the conflict.

I have no easy fix for this right now, although the current version in `master` doesn't use FastParse at all. It's still unreleased but maybe you could build it from source and test it...

**Comment by bardurdam** [comment]
I'm also experiencing this issue where FastParse 1.0 gets evicted. When will you release a new build? Are you waiting on 2.13 support?

**Comment by alonsodomin** [comment]
Yeah, I don't mind making a milestone release if this affecting several users (beyond the sporadic ammonite users)

**Comment by bardurdam** [comment]
Thanks 👍

**Comment by alonsodomin** [comment]
Version `0.6.0-M1` has been published, it should be available in Maven Central soon
